config/rbac/role.yaml

```diff
@@ -99,7 +99,24 @@ rules:
   - get
   - list
   - patch
-
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - '*'
+- apiGroups:
+  - argoproj.io
```

Comment:
This section needs to be removed from here.

Reply:
No, we need this for docker image running other wise, serviceaccount:default could not get the leases. 
This is also the newer version requirement.

Reply:
Not Leases, `argoproj.io/workflows`.

Reply:
Remove the workflow rbac from clusterrole.

Reply:
@kevdowney Do you mind share your thought why we need remove "leases", the controller would throw error "

leaderelection.go:325] error retrieving resource lock leases.coordination.k8s.io "addonmanager.keikoproj.io" is forbidden: User "system:serviceaccount:addon-manager-system:default" cannot get resource "leases"

Reply:
Not the leases, remove the workflows rbac

Reply:
I’m uncomfortable watching all the workflows in the entire cluster, there can thousands, we have a namespaced workflow-engine deployment and expect the rbac to match

Reply:
35dddd21c12b52cc2416d3133e41c3e4f51ab0aa removed

Reply:
We might have to adopt the "clusterrole" "argoproj.io" changes into the yaml file since we already restrict watching specific namespace workflows. This is also needed from controller-runtime

otherwise, controllers will fail with following error messages.
Failed to watch *v1alpha1.Workflow: failed to list *v1alpha1.Workflow: workflows.argoproj.io is forbidden: User "system:serviceaccount:addon-manager-system:default" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope